### DISCUSSION TITLE
Teslmate behind a reverse-proxy (nginx) - Grafana Error (grafana.ini root_url)

### DISCUSSION THREAD

**Discussion by LukeSkywalker993** [teslamate-org/teslamate#2711] [Q&A] [answered]
Hello everyone,
I wanted to release my TeslaMate via nginx so that I can also call it up directly in Tesla.
Unfortunately, I can't get the Grafana dashboard configured.

Since I'm brand new to Docker, I'm a bit overwhelmed here.
I found out that you have to adapt this in the grafana.ini.
______________________________________
[server]
domain = example.com
root_url = %(protocol)s://%(domain)s:%(http_port)s/grafana/
serve_from_sub_path = true
______________________________________

But if I try to change this, I get the message "access denied" in vi because it is probably a read only file.

Now I've been looking to see if I can add this line to the docker-compose.yml. I found something about this on the internet, but unfortunately it doesn't work.

My goal:
https://tesla.domain.de/ -> map: 192.168.0.1:4000 (works)
https://tesla.domain.de/dashboards -> map: 192.168.0.1:3000

______________________________________
Error Message from Grafana:
If you're seeing this Grafana has failed to load its application files

1. This could be caused by your reverse proxy settings.

2. If you host grafana under subpath make sure your grafana.ini root_url setting includes subpath. If not using a reverse proxy make sure to set serve_from_sub_path to true.

3. If you have a local dev build make sure you build frontend using: yarn start, yarn start:hot, or yarn build

4. Sometimes restarting grafana-server can help

5. Check if you are using a non-supported browser. For more information, refer to the list of [supported browsers](https://grafana.com/docs/grafana/latest/installation/requirements/#supported-web-browsers).
 ______________________________________

**Now the most important question:
Does anyone have a docker-compose.yml where I can just give the proxy settings? :-)**

I'm completely new to the Docker, Linux and Grafana world. My Docker also runs on Windows, as I am very familiar with Windows.

thanks to all!

![grafik](https://user-images.githubusercontent.com/59370744/174963940-60ae6cf8-02bb-4e69-b365-c7b22eac0a43.png)

**Comment by LukeSkywalker993** [comment, marked as answer]
i use TeslaMate on TCP 3001, not 3000!
u have to change to 3000!

Solution:

![grafik](https://user-images.githubusercontent.com/59370744/175816429-ab9b7209-287d-47df-83ad-3510c2ec8d5d.png)
![grafik](https://user-images.githubusercontent.com/59370744/175816445-7a9cfa55-d27d-4a7c-acdc-4029640f2de8.png)

![grafik](https://user-images.githubusercontent.com/59370744/175816581-675cf2ad-f8e1-4150-bb98-4f95ea8fac8a.png)


  grafana:
    image: teslamate/grafana:latest
    restart: always
    environment:
      - DATABASE_USER=teslamate
      - DATABASE_PASS=secret
      - DATABASE_NAME=teslamate
      - DATABASE_HOST=database
      - GF_SERVER_DOMAIN=tesla.domain.de
      - GF_SERVER_ROOT_URL=%(protocol)s://%(domain)s:%(http_port)s/dashboard/
      - GF_SERVER_SERVE_FROM_SUB_PATH=true

**Comment by tobiasehlert** [comment]
I am not sure how you have your installation configured, but if you have a look at the [Advanced installation with Traefik](https://docs.teslamate.org/docs/guides/traefik) guide, you should be able to adjust the configuration to match your requirement.

You could then pretty easy adjust `/grafana` to be `/dashboard` to get the URL that you are looking for in the _docker-compose.yml_.

**Reply by LukeSkywalker993** [reply]
HI, everthing is fine. It's working very good with /dashboard

**Comment by IanWardell** [comment]
Similar solution here https://github.com/adriankumpf/teslamate/issues/60#issuecomment-1633432610
